web/projects.jspf

```diff
@@ -30,22 +30,45 @@ java.util.TreeSet,
 org.opensolaris.opengrok.web.Prefix,
 org.opensolaris.opengrok.web.PageConfig,
 org.opensolaris.opengrok.configuration.Project
-"%><%
+"%><%!
+private void resetCookie(String cookiename, HttpServletRequest request, HttpServletResponse response) {
+         Cookie cookie = null;
+         Cookie[] cookies = null;
+         
+         // Get an array of Cookies associated with the this domain
+         cookies = request.getCookies();
+         
+         if( cookies != null ) {
+            
+            for (int i = 0; i < cookies.length; i++) {
+               cookie = cookies[i];
+               
+               if((cookie.getName( )).compareTo(cookiename) == 0 ) {
+                  cookie.setMaxAge(0);
+                  response.addCookie(cookie);
+               }
+            }
+          }
+}
+%><%
 /* ---------------------- projects.jspf start --------------------- */
 {
     // Use UTF-8 if no encoding is specified in the request
     if (request.getCharacterEncoding() == null) {
         request.setCharacterEncoding("UTF-8");
     }
-
-    PageConfig cfg = PageConfig.get(request);
-    cfg.getEnv().setUrlPrefix(request.getContextPath() + Prefix.SEARCH_R + "?");
-
-    String projects = cfg.getRequestedProjectsAsString();
-    if (projects.length() != 0) {
-        Cookie cookie = new Cookie(PageConfig.OPEN_GROK_PROJECT, URLEncoder.encode(projects, "utf-8"));
-        response.addCookie(cookie);
-    }
+	if (request.getParameter("hdnSearchAll") !=null && request.getParameter("hdnSearchAll").equals("true")) {
+		resetCookie(PageConfig.OPEN_GROK_PROJECT, request, response);
```

Comment:
If you follow my comment

> Also isn't better idea to move the check of is this a search all? to the getRequestedProjects and act upon there? It would make the change more transparent I think

then this part would not be necessary I think you will store all projects in the cookie or is there any problem with it?